DirectProgramming/C++SYCL_FPGA/Tutorials/Features/ip_authoring_interfaces/mmhost/README.md

```diff
@@ -0,0 +1,372 @@
+  using YProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL2>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_read});
+  using ZProps = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL3>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::dwidth<32>,
+      sycl::ext::intel::experimental::latency<1>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::read_write_mode_write});
+
+  sycl::ext::oneapi::experimental::annotated_arg<int *, XProps> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, YProps> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ZProps> z;
+
+  int size;
+
+  void operator()() const {
+#pragma unroll 4
+    for (int i = 0; i < size; i++) {
+      z[i] = x[i] + y[i];
+    }
+  }
+};
+```
+
+### Interfacing with off-chip memory
+
+If the input and output vectors are too large for on-chip memory, larger off-chip memories can be used. Consider the parameterization of a system with the following off-chip memory interfaces:
+
+* Two banks of DDR SDRAM
+* Data bus of 256 bits
+* bursts of up to 8 requests
+
+![](assets/ddr_avhost.svg)
+
+The available memory bandwidth can be better used by coalescing the 32-bit wide load-store units into wider 256-bit wide load-store units to match the memory interface.
+
+#### Example 4: A kernel that interfaces with two off-chip memories
+(Code can be found under `part4_ddr_hosts`).
+```c++
+struct DDRIP {
+  using ParamsBl1 = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL1>,
+      sycl::ext::intel::experimental::maxburst<8>,
+      sycl::ext::intel::experimental::dwidth<256>,
+      sycl::ext::oneapi::experimental::alignment<kAlignment>,
+      sycl::ext::intel::experimental::awidth<32>,
+      sycl::ext::intel::experimental::latency<0>});
+
+  using ParamsBl2 = decltype(sycl::ext::oneapi::experimental::properties{
+      sycl::ext::intel::experimental::buffer_location<kBL2>,
+      sycl::ext::intel::experimental::maxburst<8>,
+      sycl::ext::intel::experimental::dwidth<256>,
+      sycl::ext::oneapi::experimental::alignment<32>,
+      sycl::ext::intel::experimental::awidth<kAlignment>,
+      sycl::ext::intel::experimental::latency<0>});
+
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ParamsBl1> x;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ParamsBl1> y;
+  sycl::ext::oneapi::experimental::annotated_arg<int *, ParamsBl2> z;
+  int size;
+
+  void operator()() const {
+#pragma unroll 4
```

Comment:
@whitepau This should be unroll by 8 no?

Reply:
We should also state how since the `dwidth` is 256 bits, i.e., 32 bytes, we choose the unroll factor 8 (32 bytes / 4 (bytes per integer)) so that the compiler may coalesce 8 memory accesses into a single coalesced 32 byte wide access.

Reply:
Looks like it should be 8 to me. The section before even mentioned bursts of up to 8 requests